src/Type/Php/DateIntervalFormatDynamicReturnTypeExtension.php

```diff
@@ -77,15 +80,42 @@ public function getTypeFromMethodCall(MethodReflection $methodReflection, Method
 			if (strtoupper($value) === $value) {
 				$accessories[] = new AccessoryUppercaseStringType();
 			}
+			$diffInDaysType = $this->diffInDaysTypes($formatString);
 
-			if (count($accessories) === 0) {
+			if (count($accessories) === 0 && $diffInDaysType === null) {
 				return null;
 			}
 
-			$possibleReturnTypes[] = new IntersectionType([new StringType(), ...$accessories]);
+			$intersectionType = new IntersectionType([
+				new StringType(),
+				...$accessories,
+			]);
+			$possibleReturnTypes[] = $diffInDaysType === null
+				? $intersectionType
+				: new UnionType([$diffInDaysType, $intersectionType]);
```

Comment:
You don't have to add $diffInDayType in every string you could do
```
$possibleReturnTypes = [];
if ($formatString === '%a') {
     $possibleReturnTypes[] = new ConstantStringType('(unknown)');
}
```

Reply:
If the pattern is not `%a` without any additional character, it won't only return `(unknown)` and so it would only be a `lowercase-string&non-falsy-string`.